app/assets/javascripts/application.js.erb

```diff
@@ -590,6 +590,24 @@ $(document).ready(function() {
       this.value = 'http://' + this.value;
   });
 
+  // Remember story collapses; this is stored in localStorage for every story ID as an object.
+  $(document).on('change', '.comment_folder_button', function(e) {
+    var id       = $(e.target).closest('.comments_subtree').find('[data-shortid]:first').attr('data-shortid'),
```

Comment:
Could the `find` here be replaced by looking for `.closest('.comment')`?

Reply:
The way the HTML is currently layed out, no:

```
<li class="comments_subtree">
	<input id="comment_folder_inhoqk" class="comment_folder_button" type="checkbox">
	<div id="c_inhoqk" data-shortid="inhoqk" class="comment">
	    <label for="comment_folder_inhoqk" class="comment_folder">[-]</label>

	    <div class="details">
		    [..comment body and other buttons..]
	    </div>
</li>
```

`e.target` is that `input`, but I see no reason that can't be moved next to the `label` which would indeed simplify this a bit. I'll try and see if there are any problems with that.

Reply:
Actually, moving that won't work due to how the hiding is implemented: it needs to be before the `<div class="comment">` in order to hide that in CSS (no parent selector in CSS).

But it can be simplified to `find('~ div.comment')`, which matches what the CSS does.